### PR TITLE
Add tooltip for points and min_height for layout

### LINKED ISSUE #12: Add a way to set the min-height of a chart for better Pluto.jl rendering
Is your feature request related to a problem? Please describe.
---
When a chart is rendered in Pluto, it is very short by default.
image

Describe the solution you'd like
---
It would be nice if I could set `min_height="580px"` somewhere.  However, I'm not really sure where that should go, so I wanted to defer this to the maintainers.

**Honestly, a good first step might be to hard-code a value here to give people who use Pluto a good first impression of LightweightCharts.jl.**  Parametrizing the min-height can come later.

Describe alternatives you've considered
---
I currently use this hack to edit in a `min-height` style attribute before handing the chart over to Pluto for rendering.

```julia
function add_height(s; height="580px")
    replace(s, "<body>"=>"<body style=\"min-height: $(height);\">")
end
```
```julia
HTML(add_height(string(panel2); height="580px"))
```
image

Additional context
---
Thanks again for implementing `Base.show` in the first place.  #7

### LINKED ISSUE #13: Add LWC tooltips on hover
## Add  LWC tooltips on hover

Currently LightweightCharts.jl doesn’t include a built-in tooltip feature, but we can implement it. You can refer to examples from the [Lightweight Charts documentation](https://tradingview.github.io/lightweight-charts/tutorials/how_to/tooltips) for guidance.

## Requirements

The desired behaviour (or a variation of it) can be found [here](https://tradingview.github.io/lightweight-charts/tutorials/how_to/tooltips)

example_tooltip

## Tasks

The following must be completed:

- [ ] Implement a configuration interface that allows users to enable or disable tooltips for each plot separately.
- [ ] Users should be able to customize tooltip behavior (e.g., show/hide, format, position) based on their preferences.
- [ ] Document the changes thoroughly. Explain how to enable tooltips, configure them, and any additional options.
- [ ] Provide sound examples demonstrating tooltip usage in different scenarios (e.g., line charts, candlestick charts).
- [ ] Write comprehensive tests to ensure that tooltips work correctly across various chart types, data points, and interactions.

## Notice 

Any breaking changes should be discussed in advance.

### PR DESCRIPTION
**Pull request by eliza-eliza**
## Added:
- **`tooltip` parameter**:
  - Enables tooltips for points.
  - Allows users to see information when hovering over a point.
  - Closed #13.
- **`tooltip_format` parameter**:
  - Sets string formatting for tooltip text.
  - Users can display the variables `title`, `time` and `value` in the desired format.
  - Closed #13.
- **`min_height` parameter**:
  - Sets the minimum height of the Pluto.jl layout.
  - Ensures better content display, preventing the layout from being too short.
  - Closed #12.

#### Example with min_height in Pluto.jl

```Julia
using Dates
using NanoDates
using LightweightCharts

layout = lwc_layout(
    lwc_panel(
        lwc_area(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(1:500), collect(1:500));
            label_name = "lwc_area",
            line_color = "#49c7e3",
            top_color = "#74d7ed",
            bottom_color = "#85f2f000",
            line_style = LWC_SOLID,
            line_type = LWC_STEP,
            line_width = 2,
            precision = 3,
            price_scale_id = LWC_LEFT,
        );
        x = 1,
        y = 1,
    ),
    min_height = 400;
    name = "LightweightCharts ❤️ Julia"
)
```
min_height

#### Tooltip example with 10 lines

```Julia
using Dates
using NanoDates
using LightweightCharts

layout = lwc_layout(
    lwc_panel(
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(1:500), collect(1:500));
            label_name = "line1",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line2",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line3",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line4",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line5",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line6",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line7",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line8",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line9",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        lwc_line(
            NanoDate("2024-01-01") .+ Second.(1:500),
            map(x -> rand(50:1000), collect(1:500));
            label_name = "line10",
            line_style = LWC_DASHED,
            line_type = LWC_SIMPLE,
            line_width = 3,
            price_scale_id = LWC_RIGHT,
        ),
        tooltip_format = "\${title}: \${value}";
        x = 1,
        y = 1,
    );
    name = "LightweightCharts ❤️ Julia"
)

lwc_show(layout)
```
lines

#### Tooltip example with baseline

```Julia
using LightweightCharts

t_values = round(Int64, time()) .+ collect(1:500)
x_values = map(x -> sin(x / 10) + cos(x / 5), 1:500)

chart = lwc_baseline(
    t_values,
    x_values;
    label_name = "lwc_baseline",
    base_value = LWCBaseValue("price", 0.0),
    line_style = LWC_SOLID,
    line_type = LWC_CURVED,
    line_width = 3,
    precision = 4,
    price_scale_id = LWC_RIGHT,
)

lwc_show(chart)
```
baseline

#### Tooltip example with area

```Julia
using Dates
using LightweightCharts

t_values = round(Int64, time()) .+ collect(1:500)
x_values = map(x -> x % 20, 1:500)

chart = lwc_area(
    t_values,
    x_values;
    label_name = "lwc_area",
    line_color = "#00b4d8",
    top_color = "#90e0ef",
    bottom_color = "rgba(133, 242, 240, 0)",
    line_style = LWC_SOLID,
    line_type = LWC_STEP,
    line_width = 2,
    precision = 3,
    price_scale_id = LWC_RIGHT,
)

lwc_show(chart)
```
area

#### Tooltip example with histogram

```Julia
using LightweightCharts

t_values = round(Int64, time()) .+ collect(1:500)
x_values = map(x -> x * cos(x / 10), 1:500)

chart = lwc_histogram(
    t_values,
    x_values;
    label_name = "lwc_histogram",
    base = 0.0,
    color = "rgba(240, 113, 103, 0.5)",
    price_scale_id = LWC_RIGHT,
)

lwc_show(chart)
```
histogram

#### Tooltip example with candlestick

```Julia
using Dates
using LightweightCharts

open_time = now() .+ Second.(1:500)
x_values = map(x -> sin(2rand() + x / 10), 1:500)

chart = lwc_candlestick(
    LWCCandleChartItem.(
        open_time,
        x_values,
        x_values .+ rand(500),
        x_values .- rand(500),
        [x_values[2:end]..., x_values[begin]],
    );
    label_name = "lwc_candlestick",
    up_color = "#52a49a",
    down_color = "#de5e57",
    border_visible = false,
    price_scale_id = LWC_RIGHT,
)

lwc_show(chart)
```

#### Tooltip example with bar

```Julia
using Dates
using  LightweightCharts

open_time = now() .+ Second.(1:500)
x_values = map(x -> sin(2rand() + x / 10), 1:500)

chart = lwc_bar(
    LWCCandleChartItem.(
        open_time,
        x_values,
        x_values .+ rand(500),
        x_values .- rand(500),
        [x_values[2:end]..., x_values[begin]],
    );
    label_name = "lwc_bar",
    up_color = "#52a49a",
    down_color = "#de5e57",
    price_scale_id = LWC_RIGHT,
)

lwc_show(chart)
```
bar

### Additional Information:
- **Documentation:**  Updated examples demonstrating how to use the new parameters.
- **Testing:**  Tests have been updated with new parameters to ensure they work correctly.

### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [x] Did you add new tests?
- [ ] Did you add reviewers?

### CODE REVIEW

File: frontend/src/components/chart_panel.jsx [hunk 1]
```diff
@@ -91,6 +92,66 @@ const ChartPanel = ({ settings, id, setPanels }) => {
...
+                if (chartRect.width - point.x - parseInt(tooltipRef.style.marginLeft) < tooltipRef.offsetWidth) {
+                    tooltipRef.style.left = `${chartRect.left + point.x - tooltipRef.offsetWidth - 10}px`;
+                } else {
+                    tooltipRef.style.left = `${chartRect.left + point.x}px`;
+                }
+                tooltipRef.style.top  = `${chartRect.top + point.y - 22}px`;
+                tooltipRef.style.display = 'inline';
+            } else {
+                tooltipRef.style.display = 'none';
+            }
```
Comment (dmitrii-doronin) [inline frontend/src/components/chart_panel.jsx]:
```suggestion
            tooltipRef.style.display = 'none';
            if (tooltipHTML) {
                tooltipRef.innerHTML  = tooltipHTML;
                if (chartRect.width - point.x - parseInt(tooltipRef.style.marginLeft) < tooltipRef.offsetWidth) {
                    tooltipRef.style.left = `${chartRect.left + point.x - tooltipRef.offsetWidth - 10}px`;
                } else {
                    tooltipRef.style.left = `${chartRect.left + point.x}px`;
                }
                tooltipRef.style.top  = `${chartRect.top + point.y - 22}px`;
                tooltipRef.style.display = 'inline';
            } 
```

File: frontend/src/components/chart_panel.jsx [hunk 2]
```diff
@@ -91,6 +92,66 @@ const ChartPanel = ({ settings, id, setPanels }) => {
...
+            const tooltipRef = tooltip.current;
+            if (!param || !param.seriesData) {
+                tooltipRef.style.display = 'none';
+                return;
+            }
+
+            let seriesData = new Array();
+            param.seriesData.forEach((data, key, map) => {
+                if (!key.options().visible) return;
+                let color = key.options().color || key.options().topFillColor1 || key.options().lineColor || key.options().upColor || key.options().borderColor;
```
Comment (dmitrii-doronin) [inline frontend/src/components/chart_panel.jsx:114]:
```suggestion
                let color = [
                    key.options().color,
                    key.options().topFillColor1,
                    key.options().lineColor,
                    key.options().upColor,
                    key.options().borderColor
                ].some(x => x);
```
What do yout think about doing it like that?

Comment (gryumov) [reply frontend/src/components/chart_panel.jsx:114]: @eliza-eliza's approach is a more common and effective way to get the first available non-null value. It seems that the suggested approach should use .find(x => x);

File: frontend/src/components/chart_panel.jsx [hunk 3]
```diff
@@ -91,6 +92,66 @@ const ChartPanel = ({ settings, id, setPanels }) => {
             },
         });
 
+        settings.tooltip && newChart.subscribeCrosshairMove((param) => {
```
Comment (dmitrii-doronin) [inline frontend/src/components/chart_panel.jsx]: Doesn't it need a sort of a timeout?

File: frontend/src/components/chart_panel.jsx (hunk 2 above)
Comment (eliza-eliza) [reply frontend/src/components/chart_panel.jsx:114]: Thanks! Yeah, using .find(x => x) works better.

File: frontend/src/components/chart_panel.jsx (hunk 3 above)
Comment (eliza-eliza) [reply frontend/src/components/chart_panel.jsx]: A timeout isn't necessary here. It doesn't impact anything.

File: frontend/src/components/chart_panel.jsx [hunk 4]
```diff
@@ -102,7 +102,13 @@ const ChartPanel = ({ settings, id, setPanels }) => {
             let seriesData = new Array();
             param.seriesData.forEach((data, key, map) => {
                 if (!key.options().visible) return;
-                let color = key.options().color || key.options().topFillColor1 || key.options().lineColor || key.options().upColor || key.options().borderColor;
+                let color = [
```
Comment (dmitrii-doronin) [inline frontend/src/components/chart_panel.jsx]: @eliza-eliza my bad. I think you were the first time but formatting threw me off.

**Review by gryumov — APPROVED** [review]

**Comment by g-gundam** [comment]
Thanks for doing this.

**Pull request merged**